content/automate/ManageIQ/Infrastructure/VM/Transform/Import.class/__methods__/create_vm_import_request.rb

```diff
@@ -14,16 +14,36 @@ def initialize(handle = $evm)
               end
 
               def main
-                validate_root_args %w(vm dialog_name dialog_provider dialog_cluster dialog_storage dialog_sparse)
+                validate_root_args(%w(vm dialog_provider dialog_cluster dialog_storage dialog_sparse))
+                if !@handle.root['dialog_tag_category'].empty? && !@handle.root['dialog_tag_name'].empty?
+                  tag = "/#{@handle.root['dialog_tag_category']}/#{@handle.root['dialog_tag_name']}"
+                  tagged_vms = @handle.vmdb(:vm).find_tagged_with(:all => tag, :ns => '/managed')
```

Comment:
@smelamud Can we use the following to limit finding tagged VMWare VMs? $evm.vmdb('ManageIQ_Providers_Vmware_InfraManager_Vm').find_tagged_with(:all => tag, :ns => '/managed')

Reply:
@tinaafitz It is limited to VMWare VMs only now. In future it may support other providers.

Reply:
@smelamud I know the feature is limited to VMWare, but the vmdb find is not limited. Any Cloud/Infrastructure VM that contains that tag would be included here.

Reply:
@tinaafitz Done.